models/modelsmodule.cpp

```diff
@@ -432,48 +366,12 @@ ModelsModule::init( SLIInterpreter* )
     "music_message_in_proxy" );
 #endif
 
-  // register synapses
-
-  /* BeginDocumentation
-     Name: static_synapse_hpc - Variant of static_synapse with low memory
-     consumption.
-
-     Description:
-     hpc synapses store the target neuron in form of a 2 Byte index instead of
-     an 8 Byte pointer. This limits the number of thread local neurons to
-     65,536. No support for different receptor types. Otherwise identical to
-     static_synapse.
-
-     SeeAlso: synapsedict, static_synapse
-  */
-  kernel()
-    .model_manager
-    .register_connection_model< StaticConnection< TargetIdentifierPtrRport > >(
-      "static_synapse" );
-  kernel()
-    .model_manager
-    .register_connection_model< StaticConnection< TargetIdentifierIndex > >(
-      "static_synapse_hpc" );
-
+  // register all connection models: once for the normal indexing type and
+  // once for the more efficient "HPC" indexing type
+  register_connection_models< TargetIdentifierPtrRport >();
+  register_connection_models< TargetIdentifierIndex >( "_hpc" );
```

Comment:
This makes me almost wonder if we shouldn't place the suffix information in the `TargetIdentifier...` classes, to make sure that they are consistent throughout. With the current setup, users could create any kind of suffix-TI combinations.

Reply:
If this is challenging to implement, I suggest to create a new issue to remind us to consider further refactoring later and for now use the code you propose here.